controllers/front/validation.php

```diff
@@ -225,9 +238,7 @@ public function postProcess()
             'currency_iso_code' => $currency->iso_code,
             'bits_payed' => 0,
             'redirect_link' => $redirect_link,
-            'timeperiod' => Configuration::get(
-                'BLOCKONOMICS_TIMEPERIOD'
-            )
+            'timeperiod' => $time_remaining
```

Comment:
The timeperiod is used as the total time in the progress bar. Setting $time_remaining to timeperiod causes the progress bar to reset on refreshes incorrectly:
![PrestaShop3](https://user-images.githubusercontent.com/39666372/113355895-bf812000-9341-11eb-83ba-98d7e5057542.png)

The time remaining should instead be passed as a new key - `'time_remaining' => $time_remaining` and timeperiod reinstated.

Reply:
Done, now the progressbar works at it should.